dll/win32/msafd/misc/dllmain.c

```diff
@@ -1825,6 +1825,50 @@ WSPAccept(
     return AcceptSocket;
 }
 
+VOID
+NTAPI
+AfdConnectAPC(PVOID ApcContext, PIO_STATUS_BLOCK IoStatusBlock, ULONG Reserved)
+{
+    PAFDCONNECTAPCCONTEXT Context = ApcContext;
+
+    TRACE("AfdConnectAPC %p %lx %lx\n", ApcContext, IoStatusBlock->Status, IoStatusBlock->Information);
+
+    if (Context->lpConnectInfo)
+        HeapFree(GlobalHeap, 0, Context->lpConnectInfo);
+    if (!Context->lpSocket)
+    {
+        HeapFree(GlobalHeap, 0, ApcContext);
+        return;
+    }
+    if (IoStatusBlock->Status == STATUS_SUCCESS)
+    {
+        Context->lpSocket->SharedData->ConnectTime =
+            (GetTickCount() - Context->lpSocket->SharedData->ConnectTime) / 1000;
+        Context->lpSocket->SharedData->State = SocketConnected;
+        Context->lpSocket->TdiConnectionHandle = (HANDLE)IoStatusBlock->Information;
+    }
+
+    /* Re-enable Async Event */
+    SockReenableAsyncSelectEvent(Context->lpSocket, FD_WRITE);
+
+    /* FIXME: THIS IS NOT RIGHT!!! HACK HACK HACK! */
```

Comment:
Do you guys (@oleg-dubinskiy and others) have an idea what would be the _right_ thing to do, if code worked better?

Reply:
I think this line comes from the original code and unrelated to the patch.
[https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/msafd/misc/dllmain.c;h=79623a142c03fac7d11d387298ec2ed8423b62f0;hb=feb67576ddf09b847c6ecf7b8fd9675ccf74e11c#l2039](https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/msafd/misc/dllmain.c;h=79623a142c03fac7d11d387298ec2ed8423b62f0;hb=feb67576ddf09b847c6ecf7b8fd9675ccf74e11c#l2039)